lib/discourse_saml/saml_omniauth_strategy.rb

```diff
@@ -36,6 +36,25 @@ def callback_phase
     end
   end
 
+  protected
+
+  def handle_response(raw_response, opts, settings)
+    super unless SiteSetting.saml_replay_protection_enabled
+
+    response = request.params["SAMLResponse"]
+    decoded_response = OneLogin::RubySaml::Response.new(response, settings: settings)
```

Comment:
Will this error out if the response is not signed correctly?

(We need to be sure that the response is genuine before writing to redis, otherwise someone could use this as a DoS vector to fill Redis with bogus keys.)

Reply:
Perhaps we should call `super` first, which does the fingerprint and `is_valid?` check? And then we could add our replay protection after that?